modules/OTAR_CRISPR.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
+import argparse
+from datetime import datetime
+import gzip
+import json
+import logging
+import sys
+
+import pandas as pd
+
+# The statisticalTestTail is inferred by the column name which is being filtered on:
+FILTER_COLUMN_MAP = {
+    'pos|fdr': 'upper tail',
+    'neg|fdr': 'lower tail',
+    'neg|p-value': 'lower tail',
+    'pos|p-value': 'upper tail'
+}
+
+class OTAR_CRISPR_study_parser(object):
+    def __init__(self, study_file: str) -> None:
+
+        self.study_file = study_file
+
+        # Read and process study table:
+        study_df = pd.read_csv(study_file, sep='\t', skiprows=[1])
+
+        self.study_df = (
+            study_df
+
+            # drop rows with no study id or data file:
+            .loc[study_df.studyId.notna() & study_df.dataFile.notna()]
+            .assign(
+                # Split mapped diseases to list:
+                diseaseFromSourceMappedId=study_df.diseases.str.replace(' ', '').str.split('|'),
+
+                # Split data files to list:
+                dataFiles=study_df.dataFile.str.replace(' ', '').str.split('|')
+            )
+        )
+
+        logging.info(f'Number of studies processed: {len(self.study_df.studyId.unique())}')
+
+    def generate_evidence(self, data_folder: str) -> None:
+        """Processing the study dataframe"""
+
+        # Looping through the studies and generating evidence:
+        # Reading all data files and filter for significant hits:
+        hits = (
+            self.study_df[['studyId', 'dataFiles', 'dataFileType', 'filterColumn', 'threshold', 'projectId']]
+            .explode('dataFiles')
+            .assign(
+                dataFile=lambda df: df.apply(lambda x: f'{data_folder}/{x["projectId"]}/{x["dataFiles"]}', axis=1)
+            )
+            # TODO: parsing the data files should be file type dependent:
+            .apply(self.parse_MAGeCK_file, axis=1)
+        )
+
+        # Concatenate all hits into one single dataframe:
+        hits_df = (
+            pd.concat(hits.to_list())
+            .reset_index(drop=True)
+        )
+
+        # Merging:
+        self.merged_dataset = (
```

Comment:
I didn't notice `self.merged_dataset` was not in the constructor. I tend to include all instance variables in `__init__`:
```
self.merged_dataset = None
```